cache/transaction_registry_redis_test.go

```diff
@@ -79,3 +79,51 @@ func TestFailRedisTransaction(t *testing.T) {
 		t.Fatalf("unexpected: transaction should curry fail reason")
 	}
 }
+
+func TestCleanupRedisTransaction(t *testing.T) {
```

Comment:
IMHO, you should do 2 tests: one on failure (the one you did) and one on completion.